kotest-assertions/kotest-assertions-konform/src/jvmTest/kotlin/com/sksamuel/kotest/tests/konform/ValidatedMatchersTest.kt

```diff
@@ -0,0 +1,61 @@
+package com.sksamuel.kotest.tests.konform
+
+import io.konform.validation.Validation
+import io.konform.validation.jsonschema.maxLength
+import io.konform.validation.jsonschema.maximum
+import io.konform.validation.jsonschema.minLength
+import io.konform.validation.jsonschema.minimum
+import io.kotest.assertions.konform.shouldBeInvalid
+import io.kotest.assertions.konform.shouldBeValid
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.collections.shouldContain
+import io.kotest.matchers.nulls.shouldNotBeNull
+import io.kotest.shouldThrow
+import java.lang.AssertionError
+
+class ValidatedMatchersTest : StringSpec({
+   val validateUser = Validation<UserProfile> {
+      UserProfile::fullName {
+         minLength(2)
+         maxLength(100)
+      }
+
+      UserProfile::age ifPresent {
+         minimum(0)
+         maximum(150)
+      }
+   }
+
+   "shouldBeValid" {
+      val validUser = UserProfile("Alice", 25)
+
+      validateUser shouldBeValid validUser
+      shouldThrow<AssertionError> {
+         validateUser shouldBeInvalid validUser
+      }
+   }
+
+   "shouldNotBeValid" {
+      val invalidUser = UserProfile("A", -1)
+
+      validateUser.shouldBeInvalid(invalidUser) { invalid ->
```

Comment:
We could use contracts rather than the lambda. Or both.

Reply:
I've thought about it, but not sure what should we use as a contract.
Validation can be reused for many inputs and we cannot say that it is invalid after the first check. The same for the input.
Correct me if I am wrong and didn't catch the idea